Assets/Src/Shaders/GridShader.shader

```diff
@@ -0,0 +1,176 @@
+        float3 normal : NORMAL;
+        float2 uv : TEXCOORD0;
+      };
+
+      struct v2f {
+        float4 pos : SV_POSITION;
+        float3 worldNormal : TEXCOORD0;
+        float3 worldPos : TEXCOORD1;
+        float2 uv : TEXCOORD2;
+      };
+
+      sampler2D _MainTex;
+      float4 _MainTex_ST;
+      fixed4 _Color;
+      fixed4 _Specular;
+      float _Gloss;
+      fixed4 _GridColor;
+      float _GridLineWidth;
+      float _GridDimension;
+      fixed4 _SecondaryGridColor;
+      float _SecondaryGridLineWidth;
+      float _SecondaryGridDimension;
+
+      v2f vert(a2v v) {
+        v2f o;
+        o.pos = UnityObjectToClipPos(v.pos);
+        o.worldNormal = UnityObjectToWorldNormal(v.normal);
+        o.worldPos = mul(unity_ObjectToWorld, v.pos).xyz;
+        o.uv = TRANSFORM_TEX(v.uv, _MainTex);
+        return o;
+      }
+
+      // Returns a grayscale rendering mask, indicating how much the grid color will be blended to
+      // the main texture.
+      fixed GetGridMask(
+          fixed4 gridColor,
+          float gridDimension,
+          float gridLineWidth,
+          float2 uv) {
+        // Calculates the rendering mask for grid lines.
+        const float Pi = 3.14159;
+        const float GridLineDensity = 3000.0;
+        // Scales a periodic curve (y = cos(x)) to the expected period and amplitude, then clips an
+        // arc out of each period of the curve. The clipped arc‘s width is not strictly proportional
+        // to the gridLineWidth argument but the result is good enough.
+        float density = GridLineDensity / gridDimension;
+        float2 curve = abs(cos(gridDimension * Pi * (uv - .5)));
+        float2 clip = saturate((curve - 1) * density / gridLineWidth + 1);
+        float blend = saturate(clip.x + clip.y);
+        // Merges the grid line rendering mask with the input alpha channel.
+        return fixed(gridColor.a * blend);
+      }
+
+      fixed4 frag(v2f i) : SV_Target {
+        // Texture and lighting calculation for a specular surface.
+        fixed3 worldNormal = normalize(i.worldNormal);
+        fixed3 worldLightDir = normalize(UnityWorldSpaceLightDir(i.worldPos));
+        fixed4 texColor = tex2D(_MainTex, i.uv);
+        fixed3 albedo = texColor.rgb * _Color.rgb;
+        fixed3 ambient = UNITY_LIGHTMODEL_AMBIENT.xyz * albedo;
+        fixed3 diffuse = _LightColor0.rgb * albedo * saturate(dot(worldNormal, worldLightDir));
+        fixed3 viewDir = normalize(UnityWorldSpaceViewDir(i.worldPos));
+        fixed3 halfDir = normalize(worldLightDir + viewDir);
+        fixed3 specular =
+            _LightColor0.rgb * _Specular.rgb * pow(saturate(dot(worldNormal, halfDir)), _Gloss);
+        fixed3 main = ambient + diffuse + specular;
+
+        // Prepares the grid masks. A grid mask is a grayscale channel to indicate how much the grid
+        // color will be blended to the main color.
+        fixed4 gridMask = GetGridMask(_GridColor, _GridDimension, _GridLineWidth, i.uv);
```

Comment:
GetGridMask returns a fixed value. No need to cast to a fixed4 value?

Reply:
Done.